core/corehttp/gateway_handler.go

```diff
@@ -570,10 +584,86 @@ func (i *gatewayHandler) serveFile(w http.ResponseWriter, req *http.Request, nam
 			ctype = "text/html"
 		}
 	}
+	// Setting explicit Content-Type to avoid mime-type sniffing on the client
+	// (unifies behavior across gateways and web browsers)
 	w.Header().Set("Content-Type", ctype)
 
+	// special fixup around redirects
 	w = &statusResponseWriter{w}
-	http.ServeContent(w, req, name, modtime, content)
+
+	http.ServeContent(w, r, name, modtime, content)
+}
+
+func (i *gatewayHandler) serveRawBlock(w http.ResponseWriter, r *http.Request, blockCid cid.Cid, contentPath ipath.Path) {
```

Comment:
:information_source:  this func implements the poc for `/ipfs/{cid}?format=block` on gateways

- Unlike `/api/v0/block/get`, URLs are safe for pasting into a browser address bar (content-disposition, content-type)
- Supports range requests which benefit nodes with poor connectivity / data limits
  - Fetch of a bigger block can be resumed from specific byte